Improve Welcome page with the focus on better onboarding.

pfefferle opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

- [x] Remove hash after opening tab.
- [x] Set #id in profiles to link to.
- [x] Plugin cards in help tab should be same height 
- [x] Keep site health always, mark it as checked when there are no critical.
- [x] Make sure step count updates with option step updates.

<img width="878" alt="Screenshot 2025-04-30 at 12 31 17" src="https://github.com/user-attachments/assets/ea492a9c-c9f1-4fbe-9b67-205e772718c3" />

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
*

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

A guided onboarding flow after plugin activation to help users make key setup decisions and understand Fediverse concepts.

</details>

pfefferle committed:
Remove widgets and recommended plugins

pfefferle committed:
Show Health check as part of the welcome page

pfefferle committed:
Add simple launchpad

pfefferle commented:
@obenland can you have a look if that is the right direction?

pfefferle committed:
phpcs fixes

obenland commented on includes/wp-admin/class-welcome-fields.php:
```diff
@@ -117,35 +100,68 @@ public static function render_welcome_intro_section() {
-			/* translators: %s is where the button HTML will be rendered. */
 			\esc_html_e(
-				'Save this bookmarklet to reply to posts on other sites from your own blog! When visiting a post on another site, click the bookmarklet to start a reply.',
+				'Every beginning is a little tough—but you’re not alone. Start by following the checklist below. Explore the documentation, fine-tune your profile settings, and visit the help section for tips on connecting your site to the fediverse. For the best experience, make sure your site is healthy and your profile info is up to date.',
```
```suggestion
				'New beginnings can feel daunting—but you’re not alone. Start by following the checklist below. Explore the documentation, fine-tune your profile settings, and visit the help section for tips on connecting your site to the fediverse. For the best experience, make sure your site is healthy and your profile info is up to date.',
```

obenland commented:
I think it is! 

A few early notes that came to mind for me:
* I wouldn't call it Launchpad, even under the hood—it's not the same and it would probably be good to not create that expectation.
* The sections are being displayed incredibly densely, maybe because we don't have actual settings in them that gives it more whitespace? Probably not an issue with future visual iterations.
* I wonder if we could make the profile information part of a checklist item, too, to only be left with the list?
* I'm sure you have more visual polish planned for the list. If not, let's plan on it.

pfefferle committed:
Update includes/wp-admin/class-welcome-fields.php

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle commented:
> I'm sure you have more visual polish planned for the list. If not, let's plan on it.

Yes :)

Just wanted to have some early feedback.

pfefferle commented:
> I wonder if we could make the profile information part of a checklist item, too, to only be left with the list?

This is the second item!?! Or do you mean inline?

obenland commented:
> This is the second item!?! Or do you mean inline?

I mean everything below the site health section, starting with Blog Profile.
Maybe Site Health could be the first section in Settings?

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland commented on includes/wp-admin/class-settings.php:
```diff
@@ -350,6 +350,9 @@ public static function settings_page() {
 				\wp_enqueue_script( 'updates' );
 				break;
+			case 'settings':
+				\update_option( 'activitypub_launchpad_settings_visited', true );
```
I wonder if this option is even needed. Could we use the `activitypub_actor_mode` option instead and see if it has a value? The task would be completed once they selected an actor mode and saved it.

pfefferle replied:
what if the user does not change it, because they are fine with the default?

obenland replied:
We could consider removing the default in the settings callback so the radio buttons are all empty. I guess it depends on whether we want them to understand it and make a conscious choice, or just consider it.

pfefferle replied:
But this would force the user to go through every step, no "work out of the box", no easy skip.

obenland replied:
Yeah, agreed.

obenland committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Make hash links work

pfefferle committed:
call it checklist instead of launchpad

pfefferle committed:
fix apostrophe

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland commented:
What do you have in mind in terms of visual improvements to how these steps are displayed? Closer to what Dave shared on P2? I'd be happy to help with making progress on this

pfefferle commented:
I have no idea tbh. Feel free to take over!

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

pfefferle commented:
Some context: The problem with the UI drafts so far, is that they were made with a launchpad in mind and we have decided to not build one, but a checklist instead. that it is why I added the checkboxes in the draft.

pfefferle commented:
and I would not move the site health check. the settings should only be for settings and the welcome screen should exactly show these informations in my opinion. kind of like a cheat sheet.

obenland commented:
How about making the Health Check stuff part of the list? Add a step to fix them if there are issues?

If all checkboxes are checked off and there are still health check issues, we could display them separately, like when they break after they were okay previously.

pfefferle commented:
I think this is a good idea. Showing the health check section only if the Checklist is done and gone and the site has still open issues.

obenland committed:
Welcome: Try new UI (#1646)

obenland committed:
Merge branch 'trunk' into change/welcome-page

pfefferle committed:
unify wording

IT IS NOT A LAUNCHPAD!!!

pfefferle committed:
Check Profile-Changes (#1650)

And add anchor

obenland committed:
Standardize box-shadow

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Checklist: Add trailing period to done items. (#1654)

obenland committed:
Checklist: Keep better track of step counts (#1651)

obenland committed:
Checklist: Mark help tab visits as done (#1655)

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

pfefferle committed:
Change hover color to a light green if step completed (#1659)

obenland committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Make dashicon work again.

Merge error

obenland committed:
Checklist: Update option in case it exists. (#1662)

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Always show plugin health (#1665)

* Always show plugin health

* Move site health to 2

It most cases this should be fine.
Moving it to No.2 makes cleaner for these cases and reflects its importance.

* Mark Site Health done at 0 issues.

obenland committed:
Checklist: Update help tab checks (#1666)

Co-authored-by: Matthias Pfefferle <pfefferle@users.noreply.github.com>

pfefferle committed:
Checklist: Improve the wording (#1669)

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Checklist: Make health check number an option (#1675)

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

pfefferle committed:
Checklist: Fix Help-Tabs (#1670)

Co-authored-by: Konstantin Obenland <obenland@gmx.de>

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

obenland committed:
Checklist: Resolve once all steps are complete (#1679)

Co-authored-by: Matthias Pfefferle <pfefferle@users.noreply.github.com>

pfefferle committed:
Merge branch 'trunk' into change/welcome-page

matticbot committed:
Add changelog

obenland reviewed (approved):
Let's get it in and iterate as needed. Huzzah!
